networkx/convert_matrix.py

```diff
@@ -215,8 +215,10 @@ def from_pandas_adjacency(df, create_using=None):
 
     A = df.values
     G = from_numpy_array(A, create_using=create_using)
-
-    nx.relabel.relabel_nodes(G, dict(enumerate(df.columns)), copy=True)
+    try:
+        nx.relabel.relabel_nodes(G, dict(enumerate(df.columns)), copy=False)
+    except nx.NetworkXUnfeasible as err:
+        nx.relabel_nodes(G, dict(enumerate(df.columns)), copy=True)
```

Comment:
When `copy=True` the relabel_nodes function does not change G. So we need to capture the output of this function call.  We can also capture the output that way even when `copy=False`. So I think we should just return the result in both cases.

Also, let's use `nx.relabel_nodes` in both calls as well.

```python
    try:
        return nx.relabel_nodes(G, dict(enumerate(df.columns)), copy=False)
    except nx.NetworkXUnfeasible as err:
        return nx.relabel_nodes(G, dict(enumerate(df.columns)), copy=True)
```